### DISCUSSION TITLE
Omnifunc not set?

### DISCUSSION THREAD

**Discussion by wtwingate** [macvim-dev/macvim#1480] [Q&A]
I'm trying to use omnifunc in a Ruby file, which should trigger code completion using Vim's standard filetype plugin. However, when I invoke omnifunc `i_C-X_C-O` MacVim responds with `E764: Option 'omnifunc' is not set`. Filetype detection, plugin, and indent are all switched on, and MacVim is correctly detecting that I'm working with a Ruby file, yet omnifunc still isn't working.

As I understand it, omnifunc should be set by the filetype plugin when it is loaded. When running a different version of Vim from the terminal it works as expected, so the problem seems to be limited to MacVim.

Is this a bug? Or does MacVim require additional configuration that I missed?

**Comment by eirnym** [comment]
Such parts like filetype plugins are not a part of responsibility of MacVim project and these files goes from vim project directly. It would be helpful if exact change could be found.

I suggest to have a look into a better completion like LSP (Language Server Protocol) plugins. I personally use YouCompleteMe.

**Reply by wtwingate** [reply]
Thank you for your help. After looking into this more, it seems that the problem is not with the ftplugin but that MacVim cannot detect my installation of Ruby. I'm using rbenv to manage my Ruby installations, but MacVim is trying to use the default version provided by Apple.

**Reply by ychin** [reply]
MacVim is not using the Ruby provided by Apple. It's using the one provided by Homebrew instead. You can tell by typing `:set rubydll?` in Vim, which should show the `libruby.dylib` in Homebrew. If you want to use your own Ruby and not have to install Homebrew's version, you can simply set `rubydll` to point to your own dylib. It should be Ruby 3.2 per MacVim r179's release notes, but the next binary of MacVim (r180) will be built against Ruby 3.3 which was released recently.
